plugin/src/processIfWorkletNode.ts

```diff
@@ -15,30 +15,41 @@ export function processIfWorkletNode(
   >,
   state: ReanimatedPluginPass
 ) {
+  let shouldBeProcessed = false;
   fun.traverse({
     DirectiveLiteral(path) {
       const value = path.node.value;
-      if (
-        value === 'worklet' &&
-        path.getFunctionParent() === fun &&
-        isBlockStatement(fun.node.body)
-      ) {
-        // make sure "worklet" is listed among directives for the fun
-        // this is necessary as because of some bug, babel will attempt to
-        // process replaced function if it is nested inside another function
-        const directives = fun.node.body.directives;
-        if (
-          directives &&
-          directives.length > 0 &&
-          directives.some(
-            (directive) =>
-              isDirectiveLiteral(directive.value) &&
-              directive.value.value === 'worklet'
-          )
+      if (value === 'worklet' && isBlockStatement(fun.node.body)) {
+        const parent = path.getFunctionParent();
+        if (parent === fun) {
+          // make sure "worklet" is listed among directives for the fun
+          // this is necessary as because of some bug, babel will attempt to
+          // process replaced function if it is nested inside another function
+          const directives = fun.node.body.directives;
+          if (
+            directives &&
+            directives.length > 0 &&
+            directives.some(
+              (directive) =>
+                isDirectiveLiteral(directive.value) &&
+                directive.value.value === 'worklet'
+            )
+          ) {
+            shouldBeProcessed = true;
+          }
+        } else if (
+          state.opts.useOnExitLogicForWorkletNodes &&
+          (parent?.isFunctionDeclaration() ||
+            parent?.isFunctionExpression() ||
+            parent?.isArrowFunctionExpression())
```

Comment:
Can we move this logic to a separate function `isWorkletizableFunction(parent)`?